### ISSUE TITLE
[Idea] Add Buff data

### ISSUE THREAD

**Issue by dauthleikr** [LeagueSharp/LeagueSharp.Data#15]
If you want to, for example, get the exact time a zhonyas ends (to time spells), you have to get the buff name for it. That causes scripts to break over time (changing names) and doesn't allow generic solutions. 
So basically knowing the impact of each buff on the CharacterState / other stats would be really cool :>

**Comment by ChewyMoon** [comment]
Why can't the developer just use the character state or the buff type to determine invulnerability? This clearly solves the problem of tentative buff names.

**Comment by dauthleikr** [comment]
True, but in some other cases, for example in the one I described above that won't do, and the name is still needed (The invulnerability state is also set with KayleR, ...)

My first idea was to check which states change when a buff is added, however the states take ~5 frames to change, which makes identifying the corresponding buff hard. I also searched for a even more dynamic solution: Check if the CanMove property is false and time spells accordingly. However, in this case you have no clue how long the CanMove will stay false -> can't time

That being said, I can totally understand if you do not want to put this stuff in L#.Data, as it's hard work and probably won't be needed very often

**Comment by ghost** [comment]
This works okayish:
https://github.com/LeagueSharp/LeagueSharp.Assemblies/blob/master/LeagueSharp.Utility/Tracker/Ability/AbilityTimers.cs#L106
